### PR TITLE
(docs): update Icon docs with examples

### LINKED ISSUE #23323: [Bug]: Icon - v9 docs are lacking available props
### Library

React v9 (@fluentui/react-components)

### System Info

N/A

### Are you reporting Accessibility issue?

no

### Reproduction

docs site

## Actual Behavior
[Icons documentation page](https://react.fluentui.dev/?path=/docs/concepts-developer-icons--page) is lacking the list of available props. User needs to open the icons repo and check readme.

## Expected Behavior
As for any other component, icons should have documented props and running examples

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by tomi-msft**
Fixes #23323

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.656 kB`<br />`51.96 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `33.359 kB`<br />`11.004 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/83b033de0fe819bc30cc437ec7c4a83dd4c6997e'>83b033de0fe819bc30cc437ec7c4a83dd4c6997e</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/Avatar-mount_base.html">1276</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/Avatar-mount.html">1250</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/Button-mount_base.html">930</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/Button-mount.html">918</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProvider-mount_base.html">1512</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProvider-mount.html">1460</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-mount_base.html">581</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-mount.html">587</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">541</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">572</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">577</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/MakeStyles-mount_base.html">1969</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/MakeStyles-mount.html">1960</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/SpinButton-mount_base.html">2329</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24768/perf-test-react-components/SpinButton-mount.html">2351</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 1]
```diff
@@ -25,8 +25,61 @@ You can import the sized icons in a similar way:
 import { AccessTime24Filled } from '@fluentui/react-icons';
 ```
 
+### List of available props
+
 Each icon also accepts `className` and `primaryFill` props for styling.
 
+| - Name -    | Description                                                                                                            | Default        |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------- | -------------- |
+| bundleIcon  | Pass in the filled and unfilled version of an icon and this will combine them into one icon                            | -              |
```
Comment (layershifter) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:34]: Is it a component prop?

File: change/@fluentui-react-components-eca314cc-3bb0-44aa-bee3-e4a10f37354c.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "(docs): update Icon docs with examples",
```
Comment (layershifter) [inline change/@fluentui-react-components-eca314cc-3bb0-44aa-bee3-e4a10f37354c.json:3]:
```suggestion
  "comment": "docs: update Icon docs with examples",
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 3]
```diff
@@ -25,8 +25,61 @@ You can import the sized icons in a similar way:
...
+| className   | Used to style the icon                                                                                                 | -              |
+| filled      | Used to determine whether the filled or unfilled version of an icon is the default when `bundleIcon` is used `boolean` | -              |
+| primaryFill | Used to change the primary fill of the icon                                                                            | `currentColor` |
+| title       | Specifies the title attribute for the svg                                                                              | -              |
+
+### Example
+
+```tsx
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
```
Comment (layershifter) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:44]:
```suggestion
```

See below ⬇️

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 4]
```diff
@@ -25,8 +25,61 @@ You can import the sized icons in a similar way:
...
+import {
+  AccessTimeFilled,
+  AccessTimeRegular,
+  bundleIcon,
+  iconFilledClassName,
+  iconRegularClassName,
+} from '@fluentui/react-icons';
+import { makeStyles } from '@griffel/react';
```
Comment (layershifter) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:52]:
```suggestion
import { makeStyles } from '@fluentui/react-components';
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 5]
```diff
@@ -25,8 +25,61 @@ You can import the sized icons in a similar way:
...
+
+const AccessTime = bundleIcon(AccessTimeFilled, AccessTimeRegular);
+const rootElement = document.getElementById('root');
+const styles = useIconStyles();
+ReactDOM.render(
+  <div className={styles.icon}>
+    <AccessTime aria-label="AccessTime" {...iconStyleProps} />
+  </div>,
+  rootElement,
+);
```
Comment (layershifter) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:80]: - `useIconStyles` is a React hook a can be called inside React components only
- Let's avoid usage of React DOM in our examples as React 18 has [new API](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis) 
- I would be careful with `aria-label` as AFAIK we don't recommend to use it everywhere on icons

```suggestion
function App() {
  const styles = useIconStyles();

  return (
    <div className={styles.icon}>
      <AccessTime aria-label="An AccessTime icon" {...iconStyleProps} />
    </div>
  )
);
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx (hunk 1 above)
Comment (tomi-msft) [reply packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:34]: `bundleIcon` is not a component prop, rather a helper method. I've moved it from the prop table

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 6]
```diff
@@ -25,7 +25,60 @@ You can import the sized icons in a similar way:
...
+| - Name -    | Description                                                                                                            | Default        |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------- | -------------- |
+| className   | Used to style the icon                                                                                                 | -              |
+| filled      | Used to determine whether the filled or unfilled version of an icon is the default when `bundleIcon` is used `boolean` | -              |
+| primaryFill | Used to change the primary fill of the icon                                                                            | `currentColor` |
+| title       | Specifies the title attribute for the svg                                                                              | -              |
+
+### Example
+
+```tsx
```
Comment (ling1726) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:42]: This example immediately jumps to the `bundleIcon` usage. There is no explanation what this example is doing and why `bundleIcon` is necessary here.

Please add some description about the problem being solved here, which is switching between fill and outline icon variants on hover

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 7]
```diff
@@ -25,7 +25,63 @@ You can import the sized icons in a similar way:
...
+
+  return (
+    <div className={styles.icon}>
+      <AccessTime aria-label="An AccessTime icon" {...iconStyleProps} />
+    </div>
+  )
+);
+```
 
 ### Fluent icons as fonts
```
Comment (ling1726) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:86]: Can you also add a code sample how to use icons as fonts ?

**Review by ling1726 — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**